prbt_hardware_support/test/integrationtests/integrationtest_operation_mode_setup.cpp

```diff
@@ -0,0 +1,184 @@
+
+//! @brief Provides the monitor_cartesian_speed service. Redirects service calls to a mock method.
+class ControllerMock
+{
+public:
+  void advertiseService();
+  MOCK_METHOD2(monitor_cartesian_speed_callback_, bool(std_srvs::SetBoolRequest&, std_srvs::SetBoolResponse&));
+
+private:
+  ros::NodeHandle nh_;
+  ros::ServiceServer monitor_cartesian_speed_srv_;
+};
+
+void ControllerMock::advertiseService()
+{
+  monitor_cartesian_speed_srv_ = nh_.advertiseService(MONITOR_CARTESIAN_SPEED_SERVICE,
+                                                      &ControllerMock::monitor_cartesian_speed_callback_,
+                                                      this);
+}
+
+/**
+ * @brief Test the operation_mode_setup_executor_node
+ *
+ * Complement the unittest of the operation_mode_setup_executor by covering the respective node.
+ */
+class OperationModeSetupTest : public testing::Test, public testing::AsyncTest
+{
+protected:
+  using OperationModes = prbt_hardware_support::OperationModes;
+  using GetSpeedOverride = pilz_msgs::GetSpeedOverride;
+
+protected:
+  //! @brief Setup and wait for all connections.
+  void initialize();
+  void advertiseAll();
+  testing::AssertionResult isSpeedOverride(const double& expected_speed_override);
+
+protected:
+  ros::NodeHandle nh_;
+  ControllerMock controller_mock_;
+  ros::Publisher operation_mode_pub_;
+  ros::ServiceClient get_speed_override_client_;
+};
+
+void OperationModeSetupTest::initialize()
+{
+  waitForNode(OPERATON_MODE_SETUP_EXECUTOR_NODE_NAME);
+  advertiseAll();
+  ros::service::waitForService(GET_SPEED_OVERRIDE_SERVICE);
+  waitForSubscriber(operation_mode_pub_);
+}
+
+void OperationModeSetupTest::advertiseAll()
+{
+  controller_mock_.advertiseService();
+  operation_mode_pub_ = nh_.advertise<OperationModes>(OPERATION_MODE_TOPIC, DEFAULT_QUEUE_SIZE, true);
+}
+
+testing::AssertionResult OperationModeSetupTest::isSpeedOverride(const double& expected_speed_override)
+{
+  if (!get_speed_override_client_.isValid())
+  {
+    get_speed_override_client_ = nh_.serviceClient<GetSpeedOverride>(GET_SPEED_OVERRIDE_SERVICE);
+  }
+
+  GetSpeedOverride get_speed_override;
+  if (!get_speed_override_client_.call(get_speed_override))
+  {
+    return testing::AssertionFailure() << "Service call failed: " << get_speed_override_client_.getService();
+  }
+  if (get_speed_override.response.speed_override != expected_speed_override)
+  {
+    return testing::AssertionFailure() << "Speed overrides do not match. Expected: " << expected_speed_override
+                                       << ", actual: " << get_speed_override.response.speed_override;
+  }
+  return testing::AssertionSuccess();
+}
+
+TEST_F(OperationModeSetupTest, testNodeInitialization)
+{
+  initialize();
+  SUCCEED();
+}
+
+using testing::_;
+using testing::DoAll;
+using testing::Field;
+using testing::Return;
+using testing::SetArgReferee;
+
+TEST_F(OperationModeSetupTest, testOperationModeAuto)
```

Comment:
This test checks that the speed monitoring is de-activated in operation mode `auto`. But neither the name nor the docu states this fact.

Reply:
Done.